commands/copr.go

```diff
@@ -0,0 +1,122 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	"github.com/lingo-reviews/lingo/commands/common"
+
+	"github.com/codegangsta/cli"
+	"github.com/lingo-reviews/lingo/util"
+)
+
+var CoprCMD = cli.Command{
+	Name:  "copr",
+	Usage: "Checkout a git pull request",
+	Description: `
+
+cd into a local git repository the pull request targets. The run:
+
+$ lingo copr <user>:<branch>
+
+This will: create a new branch; checkout the pull request; and reset any
+commits back to the point the branch forked from base.
+
+`[1:],
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "base,b",
+			Value: "master",
+			Usage: "The base branch the pull request is forked from.",
+		},
+		cli.BoolFlag{
+			Name:  "dry-run,d",
+			Usage: "Prints out what this command will do, without doing it.",
+		},
+		cli.StringFlag{
+			Name:  "host",
+			Value: "https://github.com",
+			Usage: "The remote service hosting the git repository",
+		},
+		cli.BoolFlag{
+			Name:  "fetch-all,f",
+			Usage: "Fetch all remotes before checking out pull request. This will ensure the correct fork point can be found.",
+		},
+	},
+	Action: copr,
+}
+
+func copr(c *cli.Context) {
+	cliArgs := c.Args()
+	if l := len(cliArgs); l != 1 {
+		common.OSErrf("expected one arguments, got %d", l)
+		return
+	}
+
+	repoPath, err := exec.Command("git", "rev-parse", "--show-toplevel").Output()
+	if err != nil {
+		common.OSErrf(err.Error())
+		return
+	}
+	repo := path.Base(strings.Trim(string(repoPath), "\n"))
+	var fetchAll string
+	if c.Bool("fetch-all") {
```

Comment:
Would it be nicer UI to have this on by default, otherwise user might be surprised by errors